custom_components/tahoma/tahoma_device.py

```diff
@@ -46,6 +46,8 @@ def device(self) -> Device:
     @property
     def name(self) -> str:
         """Return the name of the device."""
+        if hasattr(self, "_battery_sensor"):
```

Comment:
Instead of this condition, you can move it directly in TahomaBatterySensor.

Reply:
done